tests/wasm/chan_pgm.go

```diff
@@ -0,0 +1,18 @@
+// +build tinygo
```

Comment:
It looks like you put the actual tests in the same directory as the code that runs the tests (where the files necessary for running the tests seem to be prefixed with a `0`).

I would propose this organization for consistency with other tests:

* Move the tests itself (tests/wasm/*_pgm.go) to a subdirectory. The conventional subdirectory for this would be testdata (tests/wasm/testdata). This also means no `// +build tinygo` is necessary.
* Remove the `0` prefix. If needed, you could make sure tests are sorted together with a prefix (such as `test-pgm.go` etc).
* Maybe avoid underscores in filenames: they have a somewhat special in that they can represent build tags (unless they really should be conditionally compiled with the given build tag as a suffix). At least that's the convention I've followed elsewhere.

Take a look at the tests in the transform package for example, which puts `*_test.go` files in the same package as the actual transforms but puts input/output files in transform/testdata.